src/BCFKeyValueData.cc

```diff
@@ -392,10 +276,11 @@ Status BCFKeyValueData::import_gvcf(const DataCache* cache, const string& datase
     S(body_->db->collection("bcf", coll_bcf));
     S(body_->db->put(coll_bcf, dataset, data));
 
-    // Serialize header
-    writer.release();
-    S(BCFWriter::Open(hdr.get(), true, writer));
-    S(writer->contents(data));
+    // Serialize header into a string
+    int hdrlen;
+    char *buf;
+    BCFWriter::write_header(hdr.get(), &hdrlen, &buf);
```

Comment:
Aren't we leaking `buf` here? As a general policy let's try to keep manual memory management and unsafe pointer handling limited to individual function bodies -- i.e. please change the `BCFWriter` interface to return/output a `std::string`